### DISCUSSION TITLE
matnwb version conversion.

### DISCUSSION THREAD

**Discussion by PierreLeMerre** [NeurodataWithoutBorders/helpdesk#11] [General]
A question about matnwb versions. Once a file has been generated with an old version, is it possible to convert it to the most recent version easily? In my case I would need to go from 2.2.4.0 to 2.4.0.

**Comment by oruebel** [comment]
Thanks for the question. Just to also move my response from the Slack channel here. In PyNWB this id feasible by using ``export``:

```
input_file = 'my_original_file.nwb'
export_filename = 'exported_nwbfile.nwb' 

with NWBHDF5IO(input_filen, mode='r') as read_io:    
    read_nwbfile = read_io.read()
    # call the export method to write the modified NWBFile 
    # instance to a new file path. Thee original file is not modified         
    with NWBHDF5IO(export_filename, mode='w') as export_io:             
        export_io.export(src_io=read_io, nwbfile=read_nwbfile)
```

I don't believe MatNWB supports this right now, but if you can do the conversion with PyNWB, MatNWB will be able to read those upgraded files. MatNWB generate code directly from the schema, so it is tied much more closely to the version of the schema. PyNWB in contrast uses HDMF to more dynamically map between the schema and the UI classes on read/write, so it can use the latest UI classes also for older files. 

That being said, I'm not an expert in MatNWB, and MatNWB has surprised me more than once with being able to do something that I didn't think it supported, so lets see what the MatNWB experts have to say. @bendichter @in-vidrio

**Reply by bendichter** [reply]
yeah, this checks out. MatNWB will generate classes according to whatever schema is in the file, and we don't have tools in MATLAB to update a file.

**Reply by PierreLeMerre** [reply]
Ok. Pretty straightforward then!! Thanks.
